src/Microsoft.ML.ImageAnalytics/ImageType.cs

```diff
@@ -36,6 +38,19 @@ public override bool Equals(ColumnType other)
             return Width != tmp.Width;
         }
 
+        public override bool Equals(Object other)
+        {
+            var tmp = other as ColumnType;
+            if (tmp == null)
+                return false;
```

Comment:
More recent enhancements to C# called "pattern matching" make code of this style sometimes a bit more elegant, since you can put the declaration of the variable inside the test.

```csharp
return other is ColumnType tmp && Equals(tmp);
```

Isn't that a pretty thing? :)